xarray/tests/test_coding_strings.py

```diff
@@ -107,6 +107,35 @@ def test_CharacterArrayCoder_encode(data):
     assert_identical(actual, expected)
 
 
+@pytest.mark.parametrize(
+    'original',
+    [
+        Variable(('x',), [b'ab', b'cdef']),
+        Variable(('x',), [b'ab', b'cdef'], encoding={'char_dim_name': 'foo'})
+    ]
+)
+def test_CharacterArrayCoder_char_dim_name(original):
```

Comment:
I think this is better and getting warmer. The test improved the underlying code [here](https://github.com/pydata/xarray/pull/2896/files#diff-64dc8cc049658cbab07c1615c1580689R111). 

But I still dont see how to eliminate this logic. Putting an `or` in the assert does not seem like a better alternative, nor does only testing just one of the two possible input scenarios.... let me know if you have a better idea that I'm too blind to see.

Thanks,

Reply:
The clean way to do this would be to add a second parametrized argument for the expected value, e.g.,
```
@pytest.mark.parametrize(
    ['original', 'expected_char_dim_name'],
    [
        (Variable(('x',), [b'ab', b'cdef']), 'string4'),
        (Variable(('x',), [b'ab', b'cdef'], encoding={'char_dim_name': 'foo'}), 'char_dim_name')
    ]
)
def test_CharacterArrayCoder_char_dim_name(original, expected_char_dim_name):
     ...
```

Reply:
That is the first case of 2-variable usage in that test file... thanks for pointing it out.